### PR TITLE
[Enhancement] Add audit log for TaskRun & use ttl instead of label clean

### PR DESCRIPTION
**Pull request by Astralidea**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

1.add audit log for TaskRun sql
2.use ttl instead of previous clean strategy
The previous Task has a binding with the Label's cleanup mechanism, and now take this parameter out.
The previous cleaning strategy was more complicated, but now it is simpler to unify into TTL. In addition, the Submit task will definitely expire. If there are other types that do not set TTL, they will never be cleaned up.
3.Optimize the submission result of Submit, and prompt Reject when the concurrent high lock acquisition fails.
separate task & taskrun lock and add config for control running taskrun and pending taskrun limit.

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 16 / 96 (16.67%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/scheduler/SqlTaskRunProcessor.java|0|62|00.00%|
|:large_blue_circle:|com/starrocks/scheduler/TaskManager.java|4|19|21.05%|
|:large_blue_circle:|com/starrocks/scheduler/persist/TaskRunStatus.java|2|3|66.67%|
|:large_blue_circle:|com/starrocks/scheduler/TaskRunContext.java|2|3|66.67%|
|:large_blue_circle:|com/starrocks/scheduler/Task.java|3|4|75.00%|
|:large_blue_circle:|com/starrocks/common/Config.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/scheduler/TaskRun.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/scheduler/TaskBuilder.java|1|1|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/scheduler/TaskManager.java [hunk 1]
```diff
@@ -389,45 +388,37 @@ public void removeOldTaskInfo() {
...
+        Iterator<Task> iterator = currentTask.iterator();
         List<Long> taskIdToDelete = Lists.newArrayList();
-        currentTask.sort((o1, o2) -> Long.signum(o1.getCreateTime() - o2.getCreateTime()));
-        int labelKeepMaxSecond = Config.label_keep_max_second;
-        int numTaskToRemove = currentTask.size() - Config.label_keep_max_num;
-        for (Task task : currentTask) {
-            if ((currentTimeMs - task.getCreateTime()) / 1000 > labelKeepMaxSecond ||
-                    numTaskToRemove > 0) {
+        while (iterator.hasNext()) {
+            Task task = iterator.next();
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/scheduler/TaskManager.java:394]: any concurrent control?

Comment (Astralidea) [reply fe/fe-core/src/main/java/com/starrocks/scheduler/TaskManager.java:394]: Here is just a count of which should be deleted, the real deletion is in dropTasks and concurrent control in it.

Comment (gengjun-git) [reply fe/fe-core/src/main/java/com/starrocks/scheduler/TaskManager.java:394]: I thick there will be read/write conflict if you do not get the read lock here.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 42 / 135 (31.11%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/scheduler/SqlTaskRunProcessor.java|0|62|00.00%|
|:large_blue_circle:|com/starrocks/scheduler/TaskManager.java|25|51|49.02%|
|:large_blue_circle:|com/starrocks/scheduler/TaskRunManager.java|3|5|60.00%|
|:large_blue_circle:|com/starrocks/scheduler/persist/TaskRunStatus.java|2|3|66.67%|
|:large_blue_circle:|com/starrocks/scheduler/TaskRunContext.java|2|3|66.67%|
|:large_blue_circle:|com/starrocks/scheduler/Task.java|3|4|75.00%|
|:large_blue_circle:|com/starrocks/common/Config.java|4|4|100.00%|
|:large_blue_circle:|com/starrocks/scheduler/TaskRun.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/scheduler/TaskBuilder.java|1|1|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/scheduler/TaskManager.java [hunk 2]
```diff
@@ -388,50 +413,47 @@ public void replayDropTaskRuns(List<String> queryIdList) {
...
-        for (Task task : currentTask) {
-            if ((currentTimeMs - task.getCreateTime()) / 1000 > labelKeepMaxSecond ||
-                    numTaskToRemove > 0) {
-                taskIdToDelete.add(task.getId());
-                --numTaskToRemove;
+        if (!tryTaskLock()) {
+            return;
+        }
+        List<Task> currentTask = showTasks(null);
+        try {
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/scheduler/TaskManager.java:421]:
```suggestion
        try {
            List<Task> currentTask = showTasks(null);
```

File: fe/fe-core/src/main/java/com/starrocks/scheduler/TaskManager.java [hunk 3]
```diff
@@ -388,50 +413,47 @@ public void replayDropTaskRuns(List<String> queryIdList) {
...
 
-        Deque<TaskRunStatus> taskRunHistory = taskRunManager.getTaskRunHistory().getAllHistory();
         List<String> historyToDelete = Lists.newArrayList();
 
-        if (!tryLock()) {
+        if (!tryTaskRunLock()) {
             return;
         }
+        // only SUCCESS and FAILED in taskRunHistory
+        Deque<TaskRunStatus> taskRunHistory = taskRunManager.getTaskRunHistory().getAllHistory();
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/scheduler/TaskManager.java:444]: Put it in try block

**Comment by Astralidea** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 42 / 137 (30.66%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/scheduler/SqlTaskRunProcessor.java|0|64|00.00%|
|:large_blue_circle:|com/starrocks/scheduler/TaskManager.java|25|51|49.02%|
|:large_blue_circle:|com/starrocks/scheduler/TaskRunManager.java|3|5|60.00%|
|:large_blue_circle:|com/starrocks/scheduler/persist/TaskRunStatus.java|2|3|66.67%|
|:large_blue_circle:|com/starrocks/scheduler/TaskRunContext.java|2|3|66.67%|
|:large_blue_circle:|com/starrocks/scheduler/Task.java|3|4|75.00%|
|:large_blue_circle:|com/starrocks/common/Config.java|4|4|100.00%|
|:large_blue_circle:|com/starrocks/scheduler/TaskRun.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/scheduler/TaskBuilder.java|1|1|100.00%|

**Review by gengjun-git — DISMISSED** [review]

**Comment by Astralidea** [comment]
run starrocks_fe_unittest

**Comment by Astralidea** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 43 / 140 (30.71%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/scheduler/SqlTaskRunProcessor.java|0|64|00.00%|
|:large_blue_circle:|com/starrocks/scheduler/TaskManager.java|25|51|49.02%|
|:large_blue_circle:|com/starrocks/scheduler/TaskRunManager.java|4|8|50.00%|
|:large_blue_circle:|com/starrocks/scheduler/persist/TaskRunStatus.java|2|3|66.67%|
|:large_blue_circle:|com/starrocks/scheduler/TaskRunContext.java|2|3|66.67%|
|:large_blue_circle:|com/starrocks/scheduler/Task.java|3|4|75.00%|
|:large_blue_circle:|com/starrocks/common/Config.java|4|4|100.00%|
|:large_blue_circle:|com/starrocks/scheduler/TaskRun.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/scheduler/TaskBuilder.java|1|1|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 48 / 146 (32.88%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/scheduler/SqlTaskRunProcessor.java|0|64|00.00%|
|:large_blue_circle:|com/starrocks/scheduler/TaskManager.java|25|51|49.02%|
|:large_blue_circle:|com/starrocks/scheduler/TaskRunManager.java|4|8|50.00%|
|:large_blue_circle:|com/starrocks/scheduler/persist/TaskRunStatus.java|2|3|66.67%|
|:large_blue_circle:|com/starrocks/scheduler/TaskRunContext.java|2|3|66.67%|
|:large_blue_circle:|com/starrocks/scheduler/Task.java|3|4|75.00%|
|:large_blue_circle:|com/starrocks/scheduler/Constants.java|5|6|83.33%|
|:large_blue_circle:|com/starrocks/common/Config.java|4|4|100.00%|
|:large_blue_circle:|com/starrocks/scheduler/TaskRun.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/scheduler/TaskBuilder.java|1|1|100.00%|

**Review by wyb — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Pull request merged**